src/battle_util.c

```diff
@@ -3558,11 +3558,29 @@ u8 AtkCanceller_UnableToUseMove(u32 moveType)
         case CANCELLER_POWDER_STATUS:
             if (gBattleMons[gBattlerAttacker].status2 & STATUS2_POWDER)
             {
-                if (moveType == TYPE_FIRE)
+                u32 partnerMove = gBattleMons[BATTLE_PARTNER(gBattlerAttacker)].moves[gBattleStruct->chosenMovePositions[BATTLE_PARTNER(gBattlerAttacker)]];
+                if ((moveType == TYPE_FIRE && !gBattleStruct->pledgeMove)
+                 || (gCurrentMove == MOVE_FIRE_PLEDGE && partnerMove == MOVE_GRASS_PLEDGE)
+                 || (gCurrentMove == MOVE_GRASS_PLEDGE && partnerMove == MOVE_FIRE_PLEDGE && gBattleStruct->pledgeMove))
                 {
                     gProtectStructs[gBattlerAttacker].powderSelfDmg = TRUE;
-                    gBattleMoveDamage = GetNonDynamaxMaxHP(gBattlerAttacker) / 4;
-                    gBattlescriptCurrInstr = BattleScript_MoveUsedPowder;
+                    if (GetBattlerAbility(gBattlerAttacker) != ABILITY_MAGIC_GUARD
+                    && (B_POWDER_RAIN < GEN_7 || !IsBattlerWeatherAffected(gBattlerAttacker, B_WEATHER_RAIN_PRIMAL)))
+                        gBattleMoveDamage = GetNonDynamaxMaxHP(gBattlerAttacker) / 4;
+
+                    if (GetActiveGimmick(gBattlerAttacker) == GIMMICK_Z_MOVE
+                     && gBattleStruct->obedienceResult == OBEYS
+                     && !HasTrainerUsedGimmick(gBattlerAttacker, GIMMICK_Z_MOVE))
+                    {
+                        RecordItemEffectBattle(gBattlerAttacker, HOLD_EFFECT_Z_CRYSTAL);
+                        SetGimmickAsActivated(gBattlerAttacker, GIMMICK_Z_MOVE);
+                        gBattleScripting.battler = gBattlerAttacker;
+                        gBattlescriptCurrInstr = BattleScript_ZMoveActivatePowder;
+                    }
```

Comment:
Why does this need additional handling instead of using `CANCELLER_Z_MOVES`?

Reply:
Changed it to be handled in `CANCELLER_Z_MOVES`. It needed additional handling because in the current implementation it used `MoveUsedPowder` which didn't play the Z-move activation animation and didn't apply Z-move effects for status moves.

Reply:
I see, thx!